analysis_notebooks/Attribute Analysis.ipynb

```diff
@@ -981,6 +984,296 @@
+    "            rect = patches.Rectangle(per_0,per_w, per_h, linewidth=2,edgecolor='r',facecolor='none')\n",
+    "            ax.add_patch(rect)\n",
+    "\n",
+    "            title = 'min'\n",
+    "            if j == 1:\n",
+    "                title = 'lower quartile'\n",
+    "            elif j == 2:\n",
+    "                title = 'median'\n",
+    "            elif j == 3:\n",
+    "                title = 'upper quartile'\n",
+    "            elif j == 4:\n",
+    "                title = 'max'\n",
+    "            ax.set_title(title, fontsize=15)\n",
+    "            ax.axis(\"off\")\n",
+    "            if all_ratios[cat_index] < 0:\n",
+    "                top_hl = \"Closer\"\n",
+    "                bottom_hl = \"Further\"\n",
+    "            else:\n",
+    "                top_hl = \"Further\"\n",
+    "                bottom_hl = \"Closer\"\n",
+    "\n",
+    "            fig.suptitle(\"{} (Top, {}) vs NOT-{} (Bottom, {})\".format(attr_names[ratio_idxs[cat_index]].upper(), top_hl, attr_names[ratio_idxs[cat_index]].upper(), bottom_hl), fontsize=16)\n",
+    "            im = ax.imshow(image, alpha=.66)\n",
+    "\n",
+    "    if to_save:\n",
+    "        if np.mean(a_array) < np.mean(b_array):\n",
+    "            closer_sent = attr_names[ratio_idxs[cat_index]] + \" are closer than any other attribute.\"\n",
+    "        else:\n",
+    "            closer_sent = attr_names[ratio_idxs[cat_index]] + \" are further than any other attribute.\"\n",
+    "        to_write[5] = [\"(att_dis) Qualitative example of {0}, which has the biggest ratio in distance between object and person (which can be interpreted as a proxy for interaction) between the attributes. \".format(names) + closer_sent + \" There is a red box around the person, and blue box around the object.\"]\n",
+    "        plt.savefig(\"results/{0}/{1}/3.png\".format(folder_name, save_loc))\n",
+    "        plt.close()\n",
+    "    else:\n",
+    "        plt.show()\n",
+    "\n",
+    "def qualitative_boxexamples(cat_index):\n",
+    "    try: \n",
+    "        to_save = False\n",
+    "        if cat_index is None and first_pass:\n",
+    "            to_save = True\n",
+    "            cat_index = distance_ratio_options[0][1]\n",
+    "\n",
+    "        a = attr_distr_dists[cat_index][0]\n",
+    "        b = attr_distr_dists[cat_index][1]\n",
+    "\n",
+    "        bpa = plt.boxplot(a, sym='', widths=0.6)\n",
+    "        bpb = plt.boxplot(b, sym='', widths=0.6)\n",
+    "        plt.close()\n",
+    "\n",
+    "        save_box_images(bpa, a, attr_file_paths[cat_index][0], attr_indices[cat_index][0], \n",
+    "            bpb, b, attr_file_paths[cat_index][1], attr_indices[cat_index][1],\n",
+    "            xaxis_sig[cat_index], cat_index, to_save)\n",
+    "    except (AttributeError, FileNotFoundError): \n",
+    "        print('Some functionality not available for CoCoDatasetNoImages Class')"
+   ]
+  },
```

Comment:
in the format of these notebooks, each metric has a "setup" header and then an "analyses" header, this seems to be missing the latter